docs/block-processing/us.md

```diff
@@ -0,0 +1,359 @@
+  * Update proposal size is computed as number of bytes in serialized `UpdateProposal`.
+  * Update proposal size limit is part of `BlockVersionData`.
+  * If the size of a proposal is greater than the limit from the adopted `BlockVersionData`, the proposal is invalid.
+
+* _Attributes known check_: if `verifyAllIsKnown` (defined [here](overall.md#verifyallisknown-flag))
+is `True`, all update proposal attributes must be known.
+ 
+* _Uniqueness of proposal check_: that there is no active proposal with the same id.
+
+* _Software version check_: numeric software version of application is 1 more 
+than of last confirmed proposal for this application.
+
+* _Stake to include into block check_: sum of positive votes for this proposal is enough 
+to be included into block (at least `updateProposalThd`).
+We wouldn't like adversary stakeholders to have opportunity to spam blockchain by `UpdateProposal`.
+So everyone can send update proposal to network, but it can get into block only along with enough votes.
+
+#### Protocol version checks
+
+* _Protocol version following check_: protocol version from proposal can follow last adopted protocol version.  
+The following rules regarding major and minor versions take place:
+  * The proposed major version must be equal to or greater by `1` last adopted one.
+  * If the proposed major version is greater than last adopted one by `1`, then minor version must be `0`.
+  * If the proposed major version is equal to the last adopted one, then minor version 
+can be either same as the last adopted one or greater by `1`.
+
+  Also the following rules regarding alternative versions take place (assuming checks above passed):
+  * If `(Major, Minor)` of proposed version is equal to `(Major, Minor)` of
+last adopted version, then alternative version must be equal to alternative version of last adopted version.
+  * Otherwise `(Major, Minor)` of proposed version is lexicographically greater 
+than `(Major, Minor)` of last adopted version and in this case
+other proposed block versions with same `(Major, Minor)` are considered (let's call this set `X`):
+    * If `X` is empty, given alternative version must be `0`.
+    * Otherwise it must be in `X` or greater than maximum from `X` by `1`.
+
+* _BlockVersionModifier consistenty check_: `BlockVersionModifier` from proposal is consistent 
+with `BlockVersionData` for adopted protocol version or with `BlockVersionModifier` from _competing_ proposals.  
+There are three cases to check:
+  * Block version from proposal is equal to adopted version. 
+    In this case each field of `BlockVersionModifier` from proposal must be either `Nothing` or 
+    `Just` with the same value from `BlockVersionData` of last adopted version.
+  * Block version from proposal is equal to one from another proposal. 
+    `BlockVersionModifier`s from them must be the same. 
+     If several proposals with the same block version exist, then all `BlockVersionModifier` must be equal.
```

Comment:
You should issue new proposal with new version (either bump alt version, if another protocol behaviour isn't changed in your proposal or bump `(maj, min)` version).

So this check makes sense because it's difficult to satisfy that update proposals with the same version have the same behaviour (yep, we expect it), so we try to check at least this minimal check of equality `BlockVersionModifiers`